src/gsy_e/models/strategy/state/base_states.py

```diff
@@ -443,15 +442,21 @@ def init(self):
 
     def get_dod_energy_kJ(self, time_slot: DateTime) -> float:
         """Return depth of discharge as an energy value in kJ."""
-        return (1 - self._soc[time_slot]) * self._max_capacity_kJ
+        return (1 - self._soc[time_slot]) * self.max_capacity_kJ
 
     def get_available_energy_kJ(self, time_slot: DateTime) -> float:
         """Return the available energy stored in the tank."""
-        return self._soc.get(time_slot) * self._max_capacity_kJ
+        return self._soc.get(time_slot, 0) * self.max_capacity_kJ
```

Comment:
I think that this can be changed with the other formula that we used to calculate the capacity: `(max_temp - current_temp) * Q_specific`. Having said that, the only thing that we will improve is to avoid doing the reverse calculation here, so not really important and you can leave it as is.

Reply:
Unfortunately, your suggestion does not work, because for PCM tanks, the SOC is not following this formula.